src/Listener/JsonApiListener.php

```diff
@@ -341,6 +344,62 @@ public function beforeFind(Event $event)
         }
     }
 
+    /**
+     * Add 'sort' capability
+     *
+     * @see http://jsonapi.org/format/#fetching-sorting
+     * @param string|array $sortFields Field sort request
+     * @param \Crud\Event\Subject $subject The subject
+     * @param array $options Array of options for includes.
+     * @return void
+     */
+    protected function _sortParameter($sortFields, Subject $subject, $options)
+    {
+        if (is_string($sortFields)) {
+            $sortFields = explode(',', $sortFields);
+        }
+        $sortFields = array_filter((array)$sortFields);
+
+        $order = [];
+        $includes = $this->config('include');
+        $repository = $subject->query->repository();
+        foreach ($sortFields as $sortField) {
+            $direction = 'ASC';
+            if ($sortField[0] == '-') {
+                $direction = 'DESC';
+                $sortField = substr($sortField, 1);
+            }
+
+            if (strpos($sortField, '.') !== false) {
+                list ($include, $field) = explode('.', $sortField);
+
+                if ($include === Inflector::tableize($repository->alias())) {
+                    $order[$repository->aliasField($field)] = $direction;
+                    continue;
+                }
+
+                if (!in_array($include, $includes)) {
+                    continue;
+                }
+
+                $associations = $repository->associations();
+                foreach ($associations as $association) {
+                    $subject->query->contain([
+                        $association->alias() => [
+                            'sort' => [
+                                $association->aliasField($field) => $direction,
```

Comment:
Are we sorting _every_ table by the same field here?

Reply:
Hmm, you're right. i'll get it fixed

Reply:
Thanks 

Bug is proven in d46c8f3, fixed in ae044cc